Required picasso information

carlobeltrame opened the pull request:
Fixes #3179

carlobeltrame committed:
Add required Y+S fields to camp

Refs #3179

carlobeltrame committed:
Add basic frontend to edit additional required picasso fields

carlobeltrame committed:
Small print improvements

carlobeltrame committed:
Add Y+S logo to react print picasso

carlobeltrame committed:
The Y+S logo is different in italian

carlobeltrame committed:
Add required Y+S fields to react print picasso

carlobeltrame committed:
Add Y+S logo to nuxt print picasso

carlobeltrame committed:
Work around problems caused by 90deg rotation in landscape mode

carlobeltrame committed:
Add required information to nuxt print picasso

carlobeltrame committed:
Hide fields which don't apply to all camps by default

carlobeltrame committed:
Automatically open panels which contain nontrivial data

carlobeltrame committed:
Move legalName to profile

This field should not be publicly available to all unrelated eCamp
users. But it (like the whole profile) is available to related users, so
we can safely embed it when fetching a camp or a campCollaboration (but
not when fetching a single user).

carlobeltrame committed:
Fix linter

carlobeltrame committed:
Add API tests for embedding profile and legalName in camp and campCollaboration

carlobeltrame committed:
Fix test setup

carlobeltrame committed:
Fix linter

carlobeltrame committed:
Only render fields if present

To not waste any space

carlobeltrame committed:
Fix linter and remaining tests

carlobeltrame committed:
Fix lint again

By now there are more lint fix commits than content commits in this
PR...

carlobeltrame committed:
Fix another test

carlobeltrame commented on api/src/Entity/CampCollaboration.php:
```diff
@@ -86,7 +86,7 @@
 class CampCollaboration extends BaseEntity implements BelongsToCampInterface {
     public const ITEM_NORMALIZATION_CONTEXT = [
-        'groups' => ['read', 'CampCollaboration:Camp', 'CampCollaboration:User'],
+        'groups' => ['read', 'CampCollaboration:Camp', 'CampCollaboration:User', 'User:Profile'],
```
Revert this and create a filter by camp or collaboration or user on the profiles endpoint

carlobeltrame commented on api/src/Entity/Camp.php:
```diff
@@ -65,7 +65,7 @@
 class Camp extends BaseEntity implements BelongsToCampInterface, CopyFromPrototypeInterface {
     public const ITEM_NORMALIZATION_CONTEXT = [
-        'groups' => ['read', 'Camp:Periods', 'Period:Days', 'Camp:CampCollaborations', 'CampCollaboration:User'],
+        'groups' => ['read', 'Camp:Periods', 'Period:Days', 'Camp:CampCollaborations', 'CampCollaboration:User', 'User:Profile'],
```
Revert this and create a filter by camp or collaboration or user on the profiles endpoint

carlobeltrame commented on frontend/src/components/print/print-react/components/picasso/Picasso.jsx:
```diff
@@ -56,16 +59,28 @@ function Picasso(props) {
+          </Text>
+          <Text>{period.camp().organizer}</Text>
+          {period.camp().printYSLogoOnPicasso ? (
+            <YSLogo
```
There is a z-index issue, the white border of the table below is visible on the logo sometimes

carlobeltrame commented on print/components/PicassoChunk.vue:
```diff
@@ -138,6 +243,14 @@ export default {
   top: 320px;
+
+  .fullwidth {
+    width: 960px;
```
Print preview looks fine, but it breaks in the printed pdf (maybe due to the ecamp3 header and page number footer). Maybe try to reduce this width a little. Soon we should be able to remove the rotate 90deg hack, when the browserless chrome supports switching to landscape mode.

carlobeltrame replied:
I "fixed" it for now by reducing the page margins in the @page CSS from 15mm to 10mm.

carlobeltrame committed:
Avoid z-index issue

BacLuc commented on frontend/src/components/print/print-react/components/picasso/Categories.jsx:
```diff
@@ -0,0 +1,21 @@
+  return (
+    <View style={picassoStyles.categories}>
+      {camp.categories().items.map((category) => (
+        <View style={picassoStyles.category}>
```
don't you need a key property here?

carlobeltrame replied:
Yes, thanks

carlobeltrame committed:
Revert profile embedding, and instead properly load profiles before printing

carlobeltrame committed:
Reduce page margins to avoid picasso content spilling over on next page

carlobeltrame committed:
Implement filtered profile fetching

carlobeltrame committed:
Add key to repeated element

carlobeltrame committed:
Fix test setup

carlobeltrame committed:
Fix linter

pmattmann commented:
Landscape is super. And if we wish to merge fast - feel free.
Please check Picasso-Print in Portrait as well.

Nuxt:  Page-Break
![Screenshot 2023-04-07 110749](https://user-images.githubusercontent.com/470237/230580169-f82c4b8a-adb2-42aa-80b7-07939f122286.png)

React:  Block is missing
![Screenshot 2023-04-07 110851](https://user-images.githubusercontent.com/470237/230581014-5b1dd7e5-5c10-43f4-8f82-db1fd58a24d2.png)

carlobeltrame committed:
Fix bug in vertical positioning of schedule entries in react print

Due to this bug, schedule entries starting in the last hour of the day
would previously not be printed, because they would be assigned to the
wrong hour (the last, zero-height hour in the times array) and therefore
get a height of zero, making them invisible.

carlobeltrame committed:
Quick fix for nuxt print landscape picasso with normal to large footer

carlobeltrame committed:
Fix display of font descenders in react print day header

carlobeltrame commented:
> Nuxt: Page-Break

I adjusted the picasso height on portrait mode, so your example now should look okay. But since the vuetify calendar has a fixed height, for now it does not adjust at all depending on the rest of the page content. So in camps with two rows of categories, or camps with very long leader lists or addresses, the layout will still break. This will need a general overhaul, probably by @usu some time. For now I hope it is good enough.
 
> React: Block is missing

I found a bug, this schedule entry was always squished into zero height at the bottom of the day. Thanks for mentioning it! This happened when a schedule entry started within the last hour of the day, and it will become all the more important with a follow-up PR which I'm working on, where I plan to auto-detect the best "end of day" hour (instead of hard-coding it to 24:00)
